src/serializer/ResidualHeapSerializer.js

```diff
@@ -859,27 +859,21 @@ export class ResidualHeapSerializer {
     invariant(!(val instanceof NativeFunctionValue), "all native function values should be intrinsics");
     invariant(val instanceof ECMAScriptSourceFunctionValue);
 
-    let residualBindings = this.residualFunctionBindings.get(val);
-    invariant(residualBindings);
-
-    invariant(val instanceof ECMAScriptSourceFunctionValue);
-    let serializedBindings = {};
-    let instance: FunctionInstance = {
-      serializedBindings,
-      functionValue: val,
-      scopeInstances: new Set(),
-    };
+    let instance = this.residualFunctionInstances.get(val);
+    invariant(instance);
+    let serializedBindings = instance.serializedBindings;
 
     if (this.currentFunctionBody !== this.mainBody) instance.preludeOverride = this.currentFunctionBody;
     let delayed = 1;
     let undelay = () => {
       if (--delayed === 0) {
+        invariant(instance);
         instance.insertionPoint = this.emitter.getBodyReference();
+        // TODO move this line
```

Comment:
What does this comment mean? Can you more elaborate in the comment?

Reply:
Will be removed in next PR